addons/website_sale/static/src/js/website_sale.js

```diff
@@ -612,21 +612,62 @@ export const WebsiteSale = publicWidget.Widget.extend(VariantMixin, cartHandlerM
     _applyHashFromSearch() {
         const params =  new URL(window.location).searchParams;
         if (params.get("attrib")) {
-            const dataValueIds = [];
-            for (const attrib of [].concat(params.get("attrib"))) {
+            let dataValueIds = [];
+            const dataValueBySequence = {};
+            for (const attrib of params.getAll("attrib")) {
                 const attribSplit = attrib.split('-');
-                const attribValueSelector = `.js_variant_change[name="ptal-${attribSplit[0]}"][value="${attribSplit[1]}"]`;
+                const attribValueSelector = `.js_variant_change[data-attribute_id="${attribSplit[0]}"][data-attribute_value_id="${attribSplit[1]}"]`;
                 const attribValue = this.el.querySelector(attribValueSelector);
                 if (attribValue !== null) {
-                    dataValueIds.push(attribValue.dataset.value_id);
+                    const attribute_id = attribValue.dataset.attribute_id
+                    const sequence = attribValue.dataset.sequence
+                    if (sequence == undefined){
+                       dataValueIds.push({value_id: attribValue.dataset.value_id}) 
+                    }
+                    else if (
+                        !dataValueBySequence[attribute_id] ||
+                        dataValueBySequence[attribute_id].sequence > sequence
+                    ) {
+                        dataValueBySequence[attribute_id] = {
+                            value_id: attribValue.dataset.value_id,
+                            sequence,
+                        };
+                    }
                 }
             }
+            dataValueIds = dataValueIds.concat(Object.values(dataValueBySequence))
             if (dataValueIds.length) {
-                window.location.hash = `attr=${dataValueIds.join(',')}`;
+                window.location.hash = `attr=${dataValueIds.map(dataValue => dataValue.value_id).join(',')}`;
             }
         }
         this._applyHash();
+
+        // Update hrefs of cached elements when attributes change
+        const cachedProducts = this.el.querySelectorAll(".oe_product_image_link")
+        for (const product of cachedProducts) {
+            product.href = this._applyFiltersToLinks(product.href)
+        }
+        const cachedBreadcrumbs = this.el.querySelectorAll(".keep_filters")
+        for (const breadcrumb of cachedBreadcrumbs) {
+            breadcrumb.href = this._applyFiltersToLinks(breadcrumb.href)
+        }
```

Comment:
Why is it needed to do that in js here? Why isn't it done in the template?

Reply:
When done on the template and/or controller, the product elements are cached so the url is not update even when a refresh/rerender is triggered.
Theoretically it works on localhost but on runbot the urls are never updated